django-stubs/dispatch/dispatcher.pyi

```diff
@@ -1,16 +1,22 @@
+import threading
 from collections.abc import Callable
+from logging import Logger
 from typing import Any
+from weakref import WeakKeyDictionary
 
-NONE_ID: Any
+NONE_ID: int
 NO_RECEIVERS: Any
 
+logger: Logger
+
 class Signal:
-    receivers: Any
-    providing_args: Any
-    lock: Any
-    use_caching: Any
-    sender_receivers_cache: Any
-    def __init__(self, providing_args: list[str] = ..., use_caching: bool = ...) -> None: ...
+    receivers: list[Any]
+    lock: threading.Lock
+    use_caching: bool
+    sender_receivers_cache: WeakKeyDictionary[Any, Any] | dict[Any, Any]
```

Comment:
I think `WeakKeyDictionary` is just a runtime hack. What do you think about just `dict` or `MutableMapping`?

Reply:
Seems like a better fit indeed :)